### PR TITLE
Adding *.log file preview 

### PR DESCRIPTION
**Pull request by dom3k**
#### Summary
Add *.log file preview as text file.

### CODE REVIEW

**Comment by esethna** [comment]
Thanks @dom3k! Can you help provide an example file that's expected to preview?

**Comment by dom3k** [comment]
It is basically any text file but with extension *.log. With this change you can now preview this kind files before download it.

[example.log](https://github.com/mattermost/mattermost-webapp/files/2705002/example.log)
screen

**Comment by hanzei** [comment]
@esethna This is ready for review now.

**Comment by cpanato** [comment]
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

**Comment by cpanato** [comment]
Spintmint test server creation failed. See the logs for more information.

**Comment by dom3k** [comment]
@esethna 
Weird, it looks like the server has no permission to write send files. It is no matter what kind of file I send. I always get the same error response.

Example raw response:

{"id":"api.file.write_file_locally.create_dir.app_error","message":"Encountered an error creating the directory for the new file","detailed_error":"directory=/home/ubuntu/mattermost/data/20190111/teams/noteam/channels/5oa5x796oib39xjcbdt8nnoaaw/users/ybyjkx3njfgq5xfonk6o5bp4ge/fborbfuemfbj7ftr3z4iiw1ujw, err=mkdir data/20190111: permission denied","request_id":"grw4x1eqqtyatrrpimz1jmf1pe","status_code":500}

Are you sure that the server is  working as expected?

error

**Comment by esethna** [comment]
Thanks @dom3k, looks great! Moving to dev review

File: utils/constants.jsx [hunk 1]
```diff
@@ -601,7 +601,7 @@ export const Constants = {
...
+    CODE_TYPES: ['applescript', 'as', 'atom', 'bas', 'bash', 'boot', 'c', 'c++', 'cake', 'cc', 'cjsx', 'cl2', 'clj', 'cljc', 'cljs', 'cljs.hl', 'cljscm', 'cljx', '_coffee', 'coffee', 'cpp', 'cs', 'csharp', 'cson', 'css', 'd', 'dart', 'delphi', 'dfm', 'di', 'diff', 'django', 'docker', 'dockerfile', 'dpr', 'erl', 'ex', 'exs', 'f90', 'f95', 'freepascal', 'fs', 'fsharp', 'gcode', 'gemspec', 'go', 'groovy', 'gyp', 'h', 'h++', 'handlebars', 'hbs', 'hic', 'hpp', 'hs', 'html', 'html.handlebars', 'html.hbs', 'hx', 'iced', 'irb', 'java', 'jinja', 'jl', 'js', 'json', 'jsp', 'jsx', 'kt', 'ktm', 'kts', 'lazarus', 'less', 'lfm', 'lisp', 'log', 'lpr', 'lua', 'm', 'mak', 'matlab', 'md', 'mk', 'mkd', 'mkdown', 'ml', 'mm', 'nc', 'obj-c', 'objc', 'osascript', 'pas', 'pascal', 'perl', 'php', 'php3', 'php4', 'php5', 'php6', 'pl', 'plist', 'podspec', 'pp', 'ps', 'ps1', 'py', 'r', 'rb', 'rs', 'rss', 'ruby', 'scala', 'scm', 'scpt', 'scss', 'sh', 'sld', 'sql', 'st', 'styl', 'swift', 'tex', 'thor', 'txt', 'v', 'vb', 'vbnet', 'vbs', 'veo', 'xhtml', 'xml', 'xsl', 'yaml', 'zsh'],
```
Comment (saturninoabril) [inline utils/constants.jsx]: Thanks for arranging this order.  Can you please confirm if list are the same as the old one plus `log`?

Comment (dom3k) [reply utils/constants.jsx]: For me yes.

**Comment by hmhealey** [comment]
Thanks for the contribution, @dom3k! This PR just needs to be updated to the latest master, and then it'll be ready to be merged.

**Comment by hmhealey** [comment]
@amyblais Is this still in time for 5.8, or should we wait until that branch is cut so it'll go into 5.10 then?

**Comment by dom3k** [comment]
I made rebase. Very sorry everyone for taking so long.

**Comment by hmhealey** [comment]
@esethna Is this good for 5.8 or should we hold this for merge until after we cut the branch for that?

**Comment by hanzei** [comment]
Thank you very much @dom3k for the PR! :tada:

**Pull request merged**